edit_book.php

```diff
@@ -0,0 +1,44 @@
+<?php
+include 'database.php';
+
+if (isset($_GET['id'])) {
+    $id = $_GET['id'];
+    $query = "SELECT * FROM books WHERE id = $id";
+    $result = mysqli_query($connection, $query);
+    $book = mysqli_fetch_assoc($result);
```

Comment:
Observations: Similar issue with SQL injection vulnerability in the update query.
Recommendations: Use prepared statements to safely handle user input.

```php
$stmt = $connection->prepare("UPDATE books SET title = ?, author = ?, year = ? WHERE id = ?");
$stmt->bind_param('ssii', $title, $author, $year, $id);
$stmt->execute();
$stmt->close();
```